src/Middleware/HttpLogging/src/ResponseBufferingStream.cs

```diff
@@ -188,7 +189,27 @@ public void LogResponseBody(HttpLoggingInterceptorContext logContext)
     {
         if (_logBody)
         {
-            logContext.AddParameter("ResponseBody", GetString(_encoding!));
+            logContext.AddParameter("ResponseBody", GetStringInternal());
         }
     }
+
+    private string GetStringInternal()
+    {
+        var result = _bodyBeforeClose ?? GetString(_encoding!);
+        // Reset the value after its consumption to preserve GetString(encoding) behavior
+        _bodyBeforeClose = null;
+        return result;
+    }
+
+    public override void Close()
```

Comment:
`Stream.Dispose()` calls `Close()` which could then re-create the string just to throw it away. Maybe we should set `_logBody` to false once one of the `LogResponseBody()` methods is called.

I think the request stream impl is fine, but we should double-check and maybe add a comment about the assumption.

Reply:
Basically it's going to fetch an interned empty string (refer to [BufferingStream.cs#L57](https://github.com/dotnet/aspnetcore/blob/2edb8c84fa5e64ff7877f11e0f181f0419c039c9/src/Middleware/HttpLogging/src/BufferingStream.cs#L57)), but I agree that it's better to protect this code from this kind of opportunity. I'm going to add `_hasLogged` similar to the request stream implementation, as resetting `_logBody` to `false` feels a bit unintuitive for a future reader.